### ISSUE TITLE
Error when connecting to server with self-signed SSL.

### ISSUE THREAD

**Issue by kirkschnable** [nextcloud/desktop#705]
When connecting to a server that has a self-signed SSL certificate, the client errors out before allowing you to login to the server due to a certificate error.

### Expected behaviour
Because I checked "Trust this certificate anyway", client should proceed with the certificate trusted and be able to function normally.

### Actual behaviour
I get a Chrome-looking error screen that says ERR_INSECURE_RESPONSE.

For the record, I *can* access the server through Chrome once trusting the certificate, the issue isn't server side.

### Steps to reproduce
1. Connect to a Nextcloud server that has self-signed SSL.
2. Check "Trust this certificate anyway" and proceed to the next prompt.
Screenshot 1: https://iota.lt/d/cXxnz5CCEL/G0lKQJAB.png
Screenshot 2 (error): https://iota.lt/d/DnDUkVFytG/hpx71W9B.png

### Client configuration
Client version: 2.5.0beta1 (build 20180803)
Operating system: Windows 10 Pro
OS language: English
Installation path of client: C:\Program Files (x86)\Nextcloud

### Server configuration
Server configuration is not relevant to the issue, it works without HTTPS and also works with a valid SSL certificate, issue seems to be isolated to self-signed SSL certificates.

I have a reverse proxy for external access that terminates with a valid Let's Encrypt SSL, but within my server's private LAN I use a self-signed SSL to simplify management, which is where this particular client is.  I can mitigate this problem for now by either using HTTP (which is fine since it's local anyway) or by routing the connections through my SSL termination proxy that corrects the certificate, however this worked fine in previous versions of the client with the self-signed certificate.

**Comment by kirkschnable** [comment]
I guess this SSL certificate is expired, which may be why, but I do have older Nextcloud clients that are working fine with this certificate.  I would think that once you check that "trust this certificate anyway" checkbox, all validation other than "is this the same certificate?" should not be considered.

**Comment by alex-gee** [comment]
Same here on Mac OS

**Comment by boggle247** [comment]
Same on Linux Kubuntu 18.04 (for desktop client). Nextcloud running on Ubuntu Server 18.04 in Docker Container

**Comment by lavigne958** [comment]
I have a similar issue, my server certificate has expired, when I click on "trust anyway" I either get a chrome like error. 

I will be very happy to have a look at it. 

If anyone already has a fix please feel free to push it. 
On the mean time I check if I can fix it, or even just let "trust anyway" button work

**Comment by cdoenges** [comment]
I agree that this looks like #849.

I see the same issue on macOS with client 2.5.1final (build 20181204).

**Comment by lavigne958** [comment]
I am looking at it, I found at some point a function that seem to handle SSL error, and at this point it returns false, which leads the webview to discard the whole page.
I will keep searching for a solution.

**Comment by Pojay-hcz** [comment]
Same here with Mac OS 10.13.6. Uninstalled NC desktop client 2.5.1.20181204. Installed NC desktop client 2.3.3.84 instead, works again without a problem.

**Comment by AnthropoceneRich** [comment]
If you're on a Mac then downgrade to 2.33 and install that version, it should install (it did for me). Make sure you delete the current version first. Once you've set up the older version (your login details) you can reinstall 2.5.*. But first of all you need to move the files in 'Library/Application Support/Nextcloud' to 'Library/Preferences/Nextcloud'. 

It should work fine on installation. I should add that I did this on a new installation. I don't know if it may or may not cause problems down the track.

**Comment by user23498723452** [comment]
Is this critical issue ever going to be solved?  I have been unable to use anything but an old client version because of this for 6+ months.  (Ubuntu user)

**Comment by lavigne958** [comment]
a work around as been pulled into master branch:
https://github.com/nextcloud/desktop/pull/1071

it is under the branch **stable-2.5.3-rc1**

**Comment by redm123** [comment]
I guess I have the same issue. Using a self-signed certificate for my local lan instance of Nextcloud. 
Used HTTP before, which worked. Using HTTPS with Letsencrypt on my root server instance. 
I had to turn on SSL locally, though, as apparently this is the only way to get the OS X Contacts app to connect through CardDAV.

First I don't get why the Nextcloud client complains at all, as the self singed certificate is in my local System KeyChain and HTTPS browser access works just fine. First "The host name did not match any of the valid hosts for this certificate" and if I accept and connect anyway I get a second "Invalid certificate detected - The host "..." provided an invalid certitiface. Continue?" And even though I accept both I'm unable to login. Neither normal login nor app token. All I get is an endless loop of login and grant access... With app token it ends with a hanging empty Connection Wizard window.

**Comment by rucomcode** [comment]
> I guess I have the same issue. Using a self-signed certificate for my local lan instance of Nextcloud.
> Used HTTP before, which worked. Using HTTPS with Letsencrypt on my root server instance.
> I had to turn on SSL locally, though, as apparently this is the only way to get the OS X Contacts app to connect through CardDAV.
> 
> First I don't get why the Nextcloud client complains at all, as the self singed certificate is in my local System KeyChain and HTTPS browser access works just fine. First "The host name did not match any of the valid hosts for this certificate" and if I accept and connect anyway I get a second "Invalid certificate detected - The host "..." provided an invalid certitiface. Continue?" And even though I accept both I'm unable to login. Neither normal login nor app token. All I get is an endless loop of login and grant access... With app token it ends with a hanging empty Connection Wizard window.

I have the same issue

**Comment by kirkschnable** [comment]
Some food for thought @rucomcode and others having this issue... I have altered my setup and no longer need be concerned about whether this is still an issue with the client or not (so I'm not sure if it is).

I assume the reason why you lack SSL internally but use Let's Encrypt externally is because like me, you are running a reverse proxy on another server somewhere.  What I did for Nextcloud and several other internal applications is I set up another reverse proxy VM internally, I copied my outside proxy configs with slight alterations, and I rsync /etc/letsencrypt/ off of my live proxy server nightly so that my internal server has valid and up to date certificates.   Then I used internal DNS to direct all of the requests to my internal DNS names to this proxy instead of to the origin servers.

Maybe not what you'd want to do but just wanted to post this solution in case someone likes it and hadn't thought of it.
